stdin

arasouli91 opened the pull request:
User's program can now take stdin. There is an Input pane with a textarea for the input. The input needs to be in there before the program runs.

You can see in the commits that I reverted my changes from the Edge compatibility pull request, because I made the mistake of doing those changes in my forked master branch and I wasn't sure if you wanted to merge it. So, this pull request is just for stdin.

arasouli91 committed:
TextEncoder/Decoder polyfill

arasouli91 committed:
Replaced spread syntax with Object.assign()

Not sure why this was breaking compatibility with Edge, spread syntax is supposed to be supported by Edge.

arasouli91 committed:
Added short-circuit condition to catch undefined crypto

I believe the issue is crypto is a member of Window, so it is not accessible in a worker.

arasouli91 committed:
Import encoding polyfill only if undefined

Polyfill for Edge.

arasouli91 committed:
Delete encoding-indexes.js

arasouli91 committed:
Delete encoding.js

arasouli91 committed:
Defined Module['stdin'] to take user input sent to commands.run

arasouli91 committed:
Input pane. Send input into run.

arasouli91 committed:
Update process-runtime.js

christosnc commented:
Hello,

I'm trying to get stdin to work without any success..
I tried to apply the changes to the precompiled cib (gh-pages branch).
I made the changes to the process-runtime.js, process.js and index.html (instead of clang.html, because of the precompiled branch) files as noted in the "Files changed" tab.

Unfortunately I can't get stdin to work, am I missing something?

Thank you!

christosnc commented:
Checking the commits, I think there is a typo:

In the Clang.html file

Line 438: "userInput:" In front of "document.get...."
Line 493: "userInput:" Instead of "inputText:"

Changing these still can't get it to work. The emModule.stdin doesn't seem to get called by clang behind.

arasouli91 committed:
Fixed a typo

arasouli91 committed:
Fixed a typo

The property for the input text in the object that is passed to the worker should be userInput not inputText

arasouli91 commented:
Thanks for catching that. That was definitely an issue. 

I just copied these contents of clang.html and process-runtime.js into the files in the gh-pages branch and everything works fine. Not sure why it isn't working for you. Try downloading the gh-pages branch, downloading these two files from here, paste them in there, and open up clang.html. 
I am using Firefox. A recent Chrome update seems to have broken cib to an extent.

christosnc commented:
Yes, thank you, I got it working too! I don't know what the issue was.

Yeah, unfortunately chrome gives "RuntimeError: memory access out of bounds", I am using Safari.

0dminnimda commented:
no way this did not got merged ...
